File: Problems/No.242_Valid_Anagram/valid_anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        
        // parameters
        int alphaTable[26] = {0};

        // start computing
        if (s.length() != t.length()) return false;

        for (int i=0; i<s.length(); i++) {

            alphaTable[s[i]-'a']++;
            alphaTable[t[i]-'a']--;

        }

        for (int i=0; i<26; i++) if (alphaTable[i] != 0) return false;
        return true;

    }
};

File: Problems/No.242_Valid_Anagram/valid_anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """

        # parameter
        alphaTable = [0]*26

        # start computing
        if (len(s) != len(t)):
            return False
        
        for i in range(len(s)):
            alphaTable[ord(s[i])-ord('a')] += 1
            alphaTable[ord(t[i])-ord('a')] -= 1
        
        for i in range(26):
            if alphaTable[i]:
                return False

        return True
